plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -227,25 +222,39 @@ func (g *genericScheduler) Preempt(pod *v1.Pod, nodeLister algorithm.NodeLister,
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	for len(nodeToVictims) > 0 {
+
+	if len(nodeToVictims) > 0 {
+		for _, extender := range g.extenders {
+			// Pass nodeToVictims to extender to do extra preemption logics. We can expect it returns a nodeToVictimsAfterExtender struct with:
+			//   1. Subset of the given nodes after Filter.
+			//   2. Additional victims on the nodes after preemption.
+			if nodeToVictimsAfterExtender, err := extender.ProcessPreemption(pod, nodeToVictims, g.cachedNodeInfoMap, pdbs); err != nil {
+				return nil, nil, nil, err
+			} else {
+				// Write back result after extender to nodeToVictims.
+				for node := range nodeToVictims {
+					// If node exists in nodeToVictims but does not exist in nodeToVictimsAfterExtender, delete it.
+					if victimAfterExtender, ok := nodeToVictimsAfterExtender[node]; !ok {
```

Comment:
I have already mentioned elsewhere that the "key" of the map that scheduler sends to extenders and also the "key" of the map that it receives should be a "node name", not a "node pointer". Otherwise, there is no hope that we find anything in the returned map.